hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroRecord.java

```diff
@@ -184,16 +185,20 @@ public HoodieRecord<T> copy() {
 
   @Override
   public HoodieRecord wrapIntoHoodieRecordPayloadWithParams(
-      Schema recordSchema, Properties props,
+      Schema recordSchema,
+      Properties props,
       Option<Pair<String, String>> simpleKeyGenFieldsOpt,
       Boolean withOperation,
       Option<String> partitionNameOp,
       Boolean populateMetaFields,
       Option<Schema> schemaWithoutMetaFields) throws IOException {
-    IndexedRecord indexedRecord = (IndexedRecord) data.getInsertValue(recordSchema, props).get();
+    HoodieAvroRecordMerger merger = HoodieAvroRecordMerger.INSTANCE;
+    HoodieAvroIndexedRecord record = (HoodieAvroIndexedRecord) merger.merge(
```

Comment:
Let's change the merge API to
```
Option<Pair<HoodieRecord, Schema>> merge(Option<HoodieRecord> older, Schema oldSchema, Option<HoodieRecord> newer, Schema newSchema, TypedProperties props)
```
so that we pass in `Option.empty` instead of `null`.

Reply:
Yeah, will create a separate PR for this change.